### ISSUE TITLE
Smart code line length validation

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1182]
# Rule request

## Thesis

1. Validate that code line not longer than N chars. Looks like 89.5 chars is a convention for now ([one](https://charlesleifer.com/blog/new-features-planned-for-python-4-0/), [two](https://youtu.be/wf-BqAjZb8M?t=260)).
2. Allow lines longer than N chars if the part after N chars contains only a string and optional comma.
3. Maybe, even such lines should be limited. For instance, not to get longer than 120 chars. However, this part can be validated by flake8 `max-line-length` without any effort from us.

Example:

```python
d = {
  # it is ok
  'one': 'some really long message that goes beyond 90 chars',
}
```

## Reasoning

> Statements longer than 80 columns will be broken into sensible chunks, unless exceeding 80 columns significantly increases readability and does not hide information. <...> However, never break user-visible strings such as printk messages, because that breaks the ability to grep for them.
> - [linux coding style](https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst#2-breaking-long-lines-and-strings)

I see a lot of Python code that does some awful breaks to fit long text messages into the project's line limit just because. However, it creates a lot of difficulties:

1. Difficult to grep.
2. Easy to miss a space on the string breaks.
1. It doesn't make code more readable at all, even decreases readability. In most cases, I don't care if the ending of an error message goes outside of my screen.

After all, it's good to relax a rule that looks not so reasonable in the modern world:

> Go has no line length limit. Don't worry about overflowing a punched card.

**Comment by sobolevn** [comment]
Yes, there are issues with some long lines: 

- URL that are long 
- Text messages

**Comment by orsinium** [comment]
Yes, exactly. Sometimes it helps to move it into a separate var:

```python
msg = "something went really-really wrong"
raise SomeReallyEnterpriseClassicJavaNameError(msg)
```

But sometimes the message itself is too long.

**Comment by ruwaid4** [comment]
Hi again, me and my friend would like to try and take on this issue. What's the difficulty level for this one?

**Comment by sobolevn** [comment]
This one is not clear to me yet. You can select any issue from: https://github.com/wemake-services/wemake-python-styleguide/milestone/16

It would be easier 🙂

**Comment by orsinium** [comment]
[flake8-bugbear](https://github.com/PyCQA/flake8-bugbear) has `B950: line too long` rule. 

https://github.com/PyCQA/flake8-bugbear/blob/5e541a8e7036cb43c97c86295eeb2c6498980f01/bugbear.py#L44-L52

The idea is a bit similar: to relax a bit line length limit.

**Comment by orsinium** [comment]
Also, if the line-length limit is violated by `# noqa...` or `# pragma...` comment. You can't just move it on a separate line, the linter says where the comment must be put.

**Comment by orsinium** [comment]
The related task for a go linter: https://github.com/walle/lll/issues/11

**Comment by orsinium** [comment]
I decided to experiment with it a bit and made a separate small plugin:
https://github.com/orsinium-labs/flake8-length

**Comment by sobolevn** [comment]
Awesome! 👍
